### DISCUSSION TITLE
Assets in custom components are not included in build

### DISCUSSION THREAD

**Discussion by szarkans** [vuejs/vitepress#4368] [Q&A] [answered]
I have some components from PrimeVue like Card, that uses `image` in header, but those images are not included in site build for some reason. Is there something i have to do?

Components are perfectly working, globally registered, `src` link is not dead, `/assets/` is **not** in public folder, it's in root folder.

```vue
<Card style="width: 25rem; overflow: hidden" class="m-0">
    <template #header>
        <Image alt="user header" src="/assets/info/donate/chainsawman.png" preview />
    </template>
</Card>
```

**Comment by brc-dd** [comment]
https://github.com/vitejs/vite-plugin-vue/tree/main/packages/plugin-vue#asset-url-handling

try something like this, but I'm not sure if it works with custom components

```ts
// .vitepress/config.ts

export default defineConfig({
  vue: {
    template: {
      transformAssetUrls: {
        Image: ['src']
      }
    }
  }
})
```

If it doesn't work you'll need to copy your assets directory to dist after building. You can either do that manually or adjust command for build or use our buildEnd hook.

**Reply by szarkans** [reply]
Now markdown's files (`![alt](src)`) not working, but component's are...

Why do vitepress has different outputs in dev mode, build-preview mode and on deploy...........

**Reply by szarkans** [reply]
Well, i moved my `assets` at `public` folder for now. I think it's not what i have to do, but it's working for now.
Is it really that nobody uses Vue components in theirs VitePress docs? Am i the only one with this problem?

**Reply by RoadsideParty** [reply]
same issue

**Reply by brc-dd** [reply, marked as answer]
> Now markdown's files (`![alt](src)`) not working, but component's are...

ah because that removed the default options

```ts
export default defineConfig({
  vue: {
    template: {
      transformAssetUrls: {
        video: ['src', 'poster'],
        source: ['src'],
        img: ['src'],
        image: ['xlink:href', 'href'],
        use: ['xlink:href', 'href'],
        Image: ['src']
      }
    }
  }
})
```

**Reply by szarkans** [reply]
Is it bad to have all assets in public folder tho?

**Reply by brc-dd** [reply]
should be fine either way

**Reply by szarkans** [reply]
Thanks!
